static/js/place/dev_place_main.tsx

```diff
@@ -395,24 +395,46 @@ const RelatedPlaces = (props: {
   place: NamedTypedPlace;
   childPlaces: NamedTypedPlace[];
 }) => {
+  const [isCollapsed, setIsCollapsed] = useState(true);
   const { place, childPlaces } = props;
   if (!childPlaces || childPlaces.length === 0) {
     return null;
   }
+  const NUM_PLACES = 15;
+  const showToggle = childPlaces.length > NUM_PLACES;
+  const toggleShowMore = () => {
+    setIsCollapsed(!isCollapsed);
+  };
+
   return (
     <div className="related-places">
       <div className="related-places-callout">Places in {place.name}</div>
       <div className="item-list-container">
         <div className="item-list-inner">
-          {childPlaces.map((place) => (
-            <div key={place.dcid} className="item-list-item">
-              <a className="item-list-text" href={`/dev-place/${place.dcid}`}>
-                {place.name}
-              </a>
-            </div>
-          ))}
+          {childPlaces
```

Comment:
I might argue this is starting to add a lot of logic within the render block. 
could we maybe store the slice in a different variable called trimmedChildPlaces and then just toggle between trimmedChildPlaces and allChildPlaces?

Reply:
Good call out. Stored the slice in a different variable and now just toggle between the lists which looks much cleaner now.